charts/astronomer/templates/houston/worker/houston-worker-deployment.yaml

```diff
@@ -0,0 +1,111 @@
+####################
+## Houston Worker ##
+####################
+{{- if .Values.houston.worker.enabled }}
+kind: Deployment
+apiVersion: apps/v1
+metadata:
+  name: {{ .Release.Name }}-houston-worker
+  labels:
+    component: houston
+    tier: astronomer
+    release: {{ .Release.Name }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    heritage: {{ .Release.Service }}
+spec:
+  replicas: {{ .Values.houston.worker.replicas }}
+  selector:
+    matchLabels:
+      tier: astronomer
+      component: houston
+      release: {{ .Release.Name }}
+  template:
+    metadata:
+      labels:
+        tier: astronomer
+        component: houston
+        release: {{ .Release.Name }}
+      annotations:
+        checksum/houston-config: {{ include (print $.Template.BasePath "/houston/api/houston-configmap.yaml") . | sha256sum }}
+    spec:
+      nodeSelector:
+{{ toYaml (default .Values.global.platformNodePool.nodeSelector .Values.nodeSelector) | indent 8 }}
+      affinity:
+{{ toYaml (default .Values.global.platformNodePool.affinity .Values.affinity) | indent 8 }}
+      tolerations:
+{{ toYaml (default .Values.global.platformNodePool.tolerations .Values.tolerations) | indent 8 }}
+      restartPolicy: Always
+      {{- if and (not .Values.houston.airflowBackendSecretName) (not .Values.houston.airflowBackendConnection) }}
+      serviceAccountName: {{ template "houston.bootstrapperServiceAccount" . }}
+      initContainers:
+        - name: wait-for-db
+          image: {{ template "houston.image" . }}
+          imagePullPolicy: {{ .Values.images.houston.pullPolicy }}
+          command:
+            - "/houston/bin/entrypoint"
+          args:
+            - "echo"
+            - "DB ready"
+          resources:
+{{ toYaml .Values.houston.resources | indent 12 }}
+          env:
+            {{- include "houston_environment" . | indent 12 }}
+        - name: wait-for-nats-server
+          command:
+            - "dockerize"
+          args:
+            - "-wait"
+            - "tcp://astronomer-nats:4222"
+            - "-timeout"
+            - "1m"
+          image: jwilder/dockerize
```

Comment:
@andriisoldatenko what about this image? Do we need it?

Reply:
AFAIK we don't have generic way for waiting for resource... so i like this tools for simplicity
can we add it to vendored? https://github.com/jwilder/dockerize
cc @schnie

Reply:
@schnie PTAL https://github.com/astronomer/ap-base/pull/8

Reply:
@schnie fixed in af256aaeb78099e3c6d59bcec2932757af24ab2e